### PR TITLE
JSON_ERROR_SYNTAX fix for json_decode(null) in PHP 7

### PR DESCRIPTION
**Pull request by fps01**
json_decode produce error when decoding `null` or empty string in PHP 7.

Example.

``` php
var_dump(json_decode(null));
var_dump(json_last_error());
```

PHP 5.4.0 - 5.6.19, hhvm-3.6.1 - 3.12.0 returns:

```
NULL
int(0)
```

PHP 7.0.0 - 7.0.4 returns:

```
NULL
int(4)
```

I think we must exclude `null` and epmty string from a decode function becouse json_decode returns a right value.
What do you think about it?

### CODE REVIEW

File: framework/helpers/BaseJson.php [hunk 1]
```diff
@@ -90,6 +90,8 @@ public static function decode($json, $asArray = true)
     {
         if (is_array($json)) {
             throw new InvalidParamException('Invalid JSON data.');
+        } elseif ((string) $json === '') {
```
Comment (samdark) [inline framework/helpers/BaseJson.php]: It's better to check it explicitly with comparing to both null and empty string.

**Comment by fps01** [comment]
@samdark I split it

**Comment by SilverFire** [comment]
Could you add a CHANGELOG line, please?

**Comment by fps01** [comment]
@SilverFire Done

**Comment by samdark** [comment]
Thanks!

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +28,7 @@ Yii Framework 2 Change Log
 - Bug #11040: Check parameter 'recursive' and disable recursive copying with option 'recursive' => false in method BaseFileHelper::copyDirectory (Ni-san)
 - Chg: HTMLPurifier dependency updated to `~4.6` (samdark)
 - Chg #11071: `yii\helpers\BaseArrayHelper::isIn()` and `isTraversable()` since now throw `\yii\base\InvalidParamException` instead of `\InvalidArgumentException` (nukkumatti)
+- Bug #11125: Fixed JSON_ERROR_SYNTAX for json_decode(null) in PHP 7
```
Comment (SilverFire) [inline framework/CHANGELOG.md]: nickname is missing

Comment (samdark) [inline framework/CHANGELOG.md]: Not in https://github.com/yiisoft/yii2/commit/258175149d1dceb44c382ea34339dafff3eed107

**Comment by fps01** [comment]
@SilverFire @samdark thanks!

**Pull request closed without merging**